.github/workflows/ci.yaml

```
@@ -52,34 +114,99 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v4
+      
       - name: Setup Go
         uses: actions/setup-go@v5
         with:
           go-version: ${{ matrix.go }}
-      - name: Install dependencies
+      
+      - name: Install system dependencies
         run: |
-          which brew
           brew install \
             pkg-config \
             opus \
             libvpx \
             x264
+      
+      # Cache FFmpeg build for macOS
+      - name: Cache FFmpeg build
+        uses: actions/cache@v4
+        id: ffmpeg-cache
+        with:
+          path: |
+            pkg/codec/ffmpeg/tmp/n7.0
+          key: ffmpeg-darwin-n7.0-${{ hashFiles('pkg/codec/ffmpeg/Makefile') }}
+          restore-keys: |
+            ffmpeg-darwin-n7.0-
+      
+      - name: Check if FFmpeg libraries exist
+        id: ffmpeg-check
+        working-directory: pkg/codec/ffmpeg
+        run: |
+          if [ -f "tmp/n7.0/lib/libavcodec.a" ] && [ -f "tmp/n7.0/lib/pkgconfig/libavcodec.pc" ]; then
+            echo "ffmpeg_exists=true" >> $GITHUB_OUTPUT
+          else
+            echo "ffmpeg_exists=false" >> $GITHUB_OUTPUT
+          fi
+      
+      - name: Build FFmpeg (if not cached or incomplete)
+        if: steps.ffmpeg-cache.outputs.cache-hit != 'true' || steps.ffmpeg-check.outputs.ffmpeg_exists != 'true'
+        working-directory: pkg/codec/ffmpeg
+        run: |
+          echo "=== Starting FFmpeg build ==="
+          ls -la tmp/ 2>/dev/null || echo "tmp directory does not exist"
+          ls -la tmp/n7.0/ 2>/dev/null || echo "n7.0 directory does not exist"
+          make build-ffmpeg
+          echo "=== FFmpeg build finished ==="
+          ls -la tmp/n7.0/
+          ls -la tmp/n7.0/lib/ 2>/dev/null || echo "lib directory not found after build"
+          ls -la tmp/n7.0/include/ 2>/dev/null || echo "include directory not found after build"
+      
+      - name: Verify FFmpeg installation
+        working-directory: pkg/codec/ffmpeg
+        run: |
+          ls -la tmp/n7.0/
+          ls -la tmp/n7.0/lib/ || echo "lib directory not found"
```

`false || echo` will always success.
It should be like
```suggestion
          ls -la tmp/n7.0/lib/ || (echo "lib directory not found"; exit 1)
```